### ISSUE TITLE
`os.PathLike[str]` raises `PydanticSchemaGenerationError`

### ISSUE THREAD

**Issue by Ravencentric** [pydantic/pydantic#10099]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Pydantic seems to not handle PathLike with subtype. There seems to be a PR that already fixed this (https://github.com/pydantic/pydantic/pull/9764) already so i'm guessing it's a bug.

### Example Code

```Python
from os import PathLike

from pydantic import validate_call


@validate_call
def realpath(path: str | PathLike[str]) -> None:
    print(path)


realpath("readme.md")
```
```
Traceback (most recent call last):
  File "c:\Users\raven\Documents\GitHub\junkenv\test.py", line 6, in <module>
    @validate_call
     ^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\validate_call_decorator.py", line 67, in validate_call
    return validate(func)
           ^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\validate_call_decorator.py", line 56, in validate
    validate_call_wrapper = _validate_call.ValidateCallWrapper(function, config, validate_return, local_ns)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_validate_call.py", line 57, in __init__
    schema = gen_schema.clean_schema(gen_schema.generate_schema(function))
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 512, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 789, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 856, in match_type
    return self._callable_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 1692, in _callable_schema
    arg_schema = self._generate_parameter_schema(name, annotation, p.default, parameter_mode)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 1414, in _generate_parameter_schema
    schema = self._apply_annotations(source_type, annotations)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 1890, in _apply_annotations
    schema = get_inner_schema(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_schema_generation_shared.py", line 83, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 1871, in inner_handler
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 789, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 871, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 895, in _match_generic_type
    return self._union_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 1207, in _union_schema
    choices.append(self.generate_schema(arg))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 512, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 789, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 871, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 919, in _match_generic_type
    return self._unknown_type_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 415, in _unknown_type_schema
    raise PydanticSchemaGenerationError(
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for os.PathLike[str]. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.

If you got this error by calling handler(<some type>) within `__get_pydantic_core_schema__` then you likely need to call `handler.generate_schema(<some type>)` since we do not call `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.

For further information visit https://errors.pydantic.dev/2.8/u/schema-for-unknown-type
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.8.2
        pydantic-core version: 2.20.1
          pydantic-core build: profile=release pgo=true
                 install path: C:\Users\raven\Documents\GitHub\junkenv\.venv\Lib\site-packages\pydantic
               python version: 3.12.4 (tags/v3.12.4:8e8a4ba, Jun  6 2024, 19:30:16) [MSC v.1940 64 bit (AMD64)]
                     platform: Windows-11-10.0.22631-SP0
             related packages: pydantic-extra-types-2.9.0 typing_extensions-4.12.2
                       commit: unknown
```

**Comment by Viicos** [comment]
The PR you linked fixes this issue but this will be available in Pydantic 2.9.0.

**Comment by Ravencentric** [comment]
Oh my bad, didn't realize it's slated for 2.9.0. Thank you.
